### PR TITLE
Update doc requirements and guide to install pandoc

### PR DESCRIPTION
**Pull request by B-Step62**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/B-Step62/mlflow/pull/10348?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/10348/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 10348
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

[Pandoc](https://pandoc.org/index.html) is not required for building docs, not R docs only. This changes modify requirements and set-up scripts, as well as CONTRIBUTION.md.

The set up script is tested on both Mac and Linux (Ubuntu).

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [x] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -836,6 +829,20 @@ python setup.py bdist_wheel
 
 First, install dependencies for building docs as described in [Environment Setup and Python configuration](#environment-setup-and-python-configuration).
 
+Building documentation requires [pandoc](https://pandoc.org/index.html), and both
+the pip package and the standalone haskel project are required. Both should have been
+already installed if you used the automated env setup script
+([dev-env-setup.sh](https://github.com/mlflow/mlflow/blob/master/dev/dev-env-setup.sh)),
+but if you manually installing dependencies,
+
+- Pip package: run `pip install pandoc --upgrade`
```
Comment (harupy) [inline CONTRIBUTING.md:838]: Is this really required?

Comment (harupy) [reply CONTRIBUTING.md:838]: I don't have the python pandoc, but can build the docs.

Comment (B-Step62) [reply CONTRIBUTING.md:838]: Good catch, I just tried same and suceeded building now. Don't quite remember where it came.... tho I think it's good to remove. Supposedly the library is just python binding of it's functionaliry.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -836,6 +829,20 @@ python setup.py bdist_wheel
...
+- Haskel project: please follow [official instruction](https://pandoc.org/installing.html).
+
+Also check you are the version of your installation via `pandoc --version`. If using Mac
```
Comment (harupy) [inline CONTRIBUTING.md:841]: Any version requirement?

Comment (B-Step62) [reply CONTRIBUTING.md:841]: 2.2.1 or above, will add

File: dev/dev-env-setup.sh [hunk 3]
```diff
@@ -130,7 +130,7 @@ if [ -z "$pyenv_exist" ]; then
       sudo apt-get update -y
       sudo apt-get install -y make build-essential libssl-dev zlib1g-dev \
       libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm \
-      libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev
+      libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev pandoc \
```
Comment (harupy) [inline dev/dev-env-setup.sh:133]: what's the version installed by apt-get?

Comment (B-Step62) [reply dev/dev-env-setup.sh:133]: yea it's older than 2, just found these lines: https://github.com/mlflow/mlflow/blame/master/mlflow/R/mlflow/Dockerfile.dev#L14-L18

Comment (B-Step62) [reply dev/dev-env-setup.sh:133]: We may want to write it down that, indeed I've been using pandoc 1.19.2 in devbox without notice... (tho I ususually build on Mac)

Comment (harupy) [reply dev/dev-env-setup.sh:133]: Can we use `wget` and `dpkg` to make sure pandoc is not old?

Comment (B-Step62) [reply dev/dev-env-setup.sh]: yup, just pushed the new revision with updated script:)

File: dev/dev-env-setup.sh [hunk 4]
```diff
@@ -105,6 +105,46 @@ minor_to_micro() {
...
 
+# Check if brew is installed and install it if it isn't present
+# Note: if xcode isn't installed, this will fail.
+# $1: name of package that requires brew
+check_and_install_brew() {
+  if [ -z "$(command -v brew)" ]; then
+    echo "Homebrew is required to install $1 on MacOS. Installing in your home directory."
+    bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+  fi
+  echo "Updating brew and installing pyenv..."
```
Comment (harupy) [inline dev/dev-env-setup.sh:116]: do we need `installing pyenv`?

Comment (B-Step62) [reply dev/dev-env-setup.sh:116]: auch, will fix

**Review by harupy — APPROVED** [review]
LGTM

File: dev/dev-env-setup.sh
Comment (harupy) [inline dev/dev-env-setup.sh:1]: dev/dev-env-setup.sh is fat.

Comment (harupy) [reply dev/dev-env-setup.sh:1]: For pandoc, I'm fine with with just documenting it on `CONTRIBUTING.md`.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -836,6 +829,16 @@ python setup.py bdist_wheel
 
 First, install dependencies for building docs as described in [Environment Setup and Python configuration](#environment-setup-and-python-configuration).
 
+Building documentation requires [Pandoc](https://pandoc.org/index.html). It should have already been
+installed if you used the automated env setup script
+([dev-env-setup.sh](https://github.com/mlflow/mlflow/blob/master/dev/dev-env-setup.sh)),
+but if you are manually installing dependencies, please follow [the official instruction](https://pandoc.org/installing.html).
```
Comment (BenWilson2) [inline CONTRIBUTING.md:835]:
```suggestion
but if you are manually installing dependencies, please follow [the official instructions](https://pandoc.org/installing.html).
```

**Review by BenWilson2 — APPROVED** [review]
LGTM!

File: dev/dev-env-setup.sh
Comment (B-Step62) [reply dev/dev-env-setup.sh:1]: Agreed, I think one way is to split it to a few logical steps and create scripts for each.  Created backlog item for this.

**Pull request merged**